File: Algorithms/src/2110. Number of Smooth Descent Periods of a Stock/Code/solution.cpp
class Solution
{
public:
    long long getDescentPeriods(vector<int> &prices)
    {
        long long ans = 1; // first day is always a valid period
        long long len = 1; // length of current smooth descent

        for (int i = 1; i < prices.size(); i++)
        {
            // check if current price is exactly 1 less than previous
            if (prices[i] == prices[i - 1] - 1)
            {
                len++; // extend the descent
            }
            else
            {
                len = 1; // reset the descent
            }
            ans += len; // add all subarrays ending here
        }
        return ans;
    }
};

File: Algorithms/src/2110. Number of Smooth Descent Periods of a Stock/Code/solution.java
class Solution {
    public long getDescentPeriods(int[] prices) {
        long ans = 1; // first day
        long len = 1; // current descent length

        for (int i = 1; i < prices.length; i++) {
            if (prices[i] == prices[i - 1] - 1) {
                len++;
            } else {
                len = 1;
            }
            ans += len;
        }
        return ans;
    }
}
